src/main/webapp/js/instructorFeedbackResults.js

```diff
@@ -205,40 +205,106 @@ function updateStatsCheckBox() {
     $('input[id=statsShownCheckBox]').val($('#show-stats-checkbox').is(':checked'));
 }
 
-function toggleCollapse(e, pans) {
-    var expand = 'Expand';
-    var collapse = 'Collapse';
-    var panels = pans || $('div.panel-collapse');
-
-    if ($(e).html().trim().startsWith(expand)) {
-        var i = 0;
-        for (var idx = 0; idx < panels.length; idx++) {
-            if ($(panels[idx]).attr('class').indexOf('in') === -1) {
-                setTimeout(showSingleCollapse, 50 * i, panels[idx]);
-                i++;
-            }
-        }
-        var htmlString = $(e).html();
-        htmlString = htmlString.replace(expand, collapse);
-        $(e).html(htmlString);
-        var tooltipString = $(e).attr('data-original-title').replace(expand, collapse);
-        $(e).attr('title', tooltipString).tooltip('fixTitle').tooltip('show');
+/**
+ * Expands or collapses all panels when collapse/expand panels button is clicked.
+ * @param {DOM} expandCollapseButton - The button that was clicked to invoke {@code #expandOrCollapsePanels}.
+ * @param {DOM} panels - The panels to be expanded/collapsed. Not defined if {@code expandCollapseButton}
+ *         is collapse panels button.
+ */
+function expandOrCollapsePanels(expandCollapseButton, panels) {
+    var STRING_EXPAND = 'Expand';
+    var STRING_COLLAPSE = 'Collapse';
+    // {@code panels} is not defined when {@code expandCollapseButton} is collapse panels button. We
+    // need to define corresponding {@code targetPanels}.
+    var targetPanels = panels || $('div.panel-collapse');
+
+    var isButtonInExapandMode = $(expandCollapseButton).html().trim().startsWith(STRING_EXPAND);
+    if (isButtonInExapandMode) {
+        // The expand/collapse button on AJAX-loaded panels has id collapse-panels-button.
+        var areAjaxLoadedPanels = $(expandCollapseButton).is($('#collapse-panels-button'));
+        expandPanels(targetPanels, areAjaxLoadedPanels);
+        replaceButtonHtmlAndTooltipText(expandCollapseButton, STRING_EXPAND, STRING_COLLAPSE);
     } else {
-        var j = 0;
-        for (var k = 0; k < panels.length; k++) {
-            if ($(panels[k]).attr('class').indexOf('in') !== -1) {
-                setTimeout(hideSingleCollapse, 100 * j, panels[k]);
-                j++;
-            }
+        collapsePanels(targetPanels);
+        replaceButtonHtmlAndTooltipText(expandCollapseButton, STRING_COLLAPSE, STRING_EXPAND);
+    }
+}
+
+/**
+ * Expands all panels. Loads panel data if missing.
+ */
+function expandPanels(panels, areAjaxLoadedPanels) {
+    var BASE_TIMEOUT_UNIT_IN_MILLI_SECONDS = 50;
```

Comment:
One possible reason is that visual effect of expanding is not really shown most of the time, but collapse is shown as the collapsed panels fit in one page. I cannot think of other reasons though.